src/cpu/x64/matmul/amx_blocking_heuristics.cpp

```diff
@@ -144,6 +142,136 @@ size_t determine_tmul_size(size_t num_elements, int full_tile_size) {
     return tmul_size;
 }
 
+bool matmul_amx_blocking_params_macro_t::maybe_small_dims_heuristics(
```

Comment:
Would be great if there was a small method description with high level logic explained.